server/src/test/java/org/elasticsearch/env/NodeEnvironmentTests.java

```diff
@@ -597,6 +597,18 @@ public void testIndexCompatibilityChecks() throws IOException {
         }
     }
 
+    public void testSymlinkDataDirectory() throws Exception {
+        Path tempDir = createTempDir().toAbsolutePath();
```

Comment:
I suspect this test is going to not be enough, since the test environment is not exactly like production. I suggest a packaging test in qa/os. There should be some other symlink related tests there I think.

Reply:
Sounds good, I'll add a test there.